tests/unit_tests/data_steward/validation/main_test.py

```diff
@@ -123,22 +133,26 @@ def test_retention_checks_list_submitted_bucket_items(self):
         expected_result = bucket_items
         self.assertCountEqual(expected_result, actual_result)
 
-        # If any AOU_REQUIRED file has been updated less than 5 minutes ago, no files should be returned
+        # If any file is missing but submission is fresh, skip it
         bucket_items = self._create_dummy_bucket_items(
             within_retention,
             after_lag_time,
             file_exclusions=['observation.csv'])
 
-        bucket_items.append({
-            'name': '2018-09-01/observation.csv',
-            'timeCreated': within_retention,
-            'updated': before_lag_time
-        })
-
         actual_result = main.list_submitted_bucket_items(bucket_items)
         expected_result = []
         self.assertCountEqual(expected_result, actual_result)
 
+        # If any file is missing but submission is stale, process it
```

Comment:
If the submission is stale, shouldn't it be ignored?
I think the logic should go into this `else,` and the function should return `files_list,` an empty list.

https://github.com/all-of-us/curation/pull/1240/files#diff-43507efb5f679e0321f8eabb85c7231a109b4000e39babde951c331470dd1593R791

I might be confused again. Let me know if my understanding is correct. Thanks!

Reply:
I referred to stale here as past 3 hours since submission, but don't have a word for past retention. Maybe you are referring to stale as past retention?

Reply:
Oh OK, now I get it! Thanks for the clarification :)